### PR TITLE
feat(source) opentelemetry-log

### PR DESCRIPTION
**Pull request by caibirdme**
Too many commits in another PR, so I created the new clear one.

```yaml
sources:
  otel:
    type: opentelemetry
    address: 0.0.0.0:6788
```

* lib/opentelemetry: otel-proto & impl From Trait to convert proto type into vector::Event
* src/soruces/opentelemetry: source plugin to setup a grpc server to receive log request

### CODE REVIEW

File: lib/opentelemetry/src/convert.rs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            );
+        }
+        if !rl.log_record.span_id.is_empty() {
+            le.insert(
+                SPAN_ID_KEY,
+                Value::Bytes(Bytes::from(hex::encode(rl.log_record.span_id))),
+            );
+        }
+        le.insert(SEVERITY_TEXT_KEY, rl.log_record.severity_text);
+        le.insert(SEVERITY_NUMBER_KEY, rl.log_record.severity_number as i64);
```
Comment (StephenWakely) [inline lib/opentelemetry/src/convert.rs:130]: Severity and severity number are also optional. Can we suppress inserting if they aren't specified (0 or empty string).

File: lib/opentelemetry/src/convert.rs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            })
+            .collect::<BTreeMap<String, Value>>(),
+    )
+}
+
+impl From<ResourceLog> for Event {
+    fn from(rl: ResourceLog) -> Self {
+        let mut le = LogEvent::default();
+        if let Some(resource) = rl.resource {
+            le.insert(RESOURCE_KEY, kv_list_into_value(resource.attributes));
```
Comment (StephenWakely) [inline lib/opentelemetry/src/convert.rs:105]: Can we suppress inserting this if it is empty.

File: lib/opentelemetry/src/convert.rs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        if !rl.log_record.attributes.is_empty() {
+            le.insert(ATTRIBUTES_KEY, kv_list_into_value(rl.log_record.attributes));
+        }
+        if let Some(v) = rl.log_record.body.and_then(|av| av.value) {
+            le.insert(log_schema().message_key(), v);
+        }
+        le.insert(
+            log_schema().timestamp_key(),
+            rl.log_record.time_unix_nano as i64,
+        );
```
Comment (StephenWakely) [inline lib/opentelemetry/src/convert.rs:116]: Can we supress this if the time is == 0.

**Review by StephenWakely — CHANGES_REQUESTED** [review]
Nice work. Many thanks for submitting this! It's exciting to see us get moving into OpenTelemetry. 

I've managed to test this using the open telemetry collector and it is working nicely. I've thrown down a few initial comments, nothing major. 

If you can tidy up the PR I'll take another look shortly.

File: lib/opentelemetry/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+mod proto;
+
+#[cfg(test)]
+mod tests {
+    #[test]
+    fn it_works() {
+        let result = 2 + 2;
+        assert_eq!(result, 4);
+    }
+}
```
[the remark this answers is not in the document]
Comment (caibirdme) [reply lib/opentelemetry/src/lib.rs:16]: What does the tidy up mean exactly? I did 'git rebase master' and then this pr contains lots of other people's commit... It's out of my expectation. So, do I need to create a new clean branch to resolve these problems or just continue to commit on this branch

**Pull request closed without merging**